src/server/transaction.cc

```diff
@@ -116,6 +116,40 @@ uint32_t Transaction::PhasedBarrier::DEBUG_Count() const {
   return count_.load(memory_order_relaxed);
 }
 
+bool Transaction::SingleClaimBarrier::IsClaimed() const {
+  return claimed_.load(memory_order_relaxed);
+}
+
+bool Transaction::SingleClaimBarrier::TryClaim() {
+  return !claimed_.exchange(true, memory_order_relaxed);  // false means first means success
+}
+
+void Transaction::SingleClaimBarrier::Release() {
+  DCHECK(claimed_.load(memory_order_relaxed));
+  released_.store(true, memory_order_relaxed);
+  ec_.notify();  // release
+}
+
+cv_status Transaction::SingleClaimBarrier::Wait(time_point tp) {
+  auto cb = [this] { return released_.load(memory_order_acquire); };
```

Comment:
Same here, I have an acquire here because currently our even-count doesn't provide us with one on the fast path